### PR TITLE
refactor(core): Introduced Projectile.getEventConsumer

### PR DESCRIPTION
**Pull request by gastaldi**
Ease testing and may be a first step to support server side events in the future

### CODE REVIEW

**Comment by gastaldi** [comment]
I thought of calling it "Beacon" (as in a beacon attached to a missile to know its location). Wdyt?

**Comment by gastaldi** [comment]
Maybe a better idea is to change the MissionControl.launch method signature to accept a Projectile and a Consumer instead of adding it to Projectile. I think they should be separate things

File: base/src/main/java/io/fabric8/launcher/base/JsonUtils.java [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import static javax.json.Json.createArrayBuilder;
 import static javax.json.Json.createObjectBuilder;
 
 public final class JsonUtils {
 
+    private static final ObjectMapper mapper = new ObjectMapper();
```
Comment (ia3andy) [inline base/src/main/java/io/fabric8/launcher/base/JsonUtils.java]: What do you think about naming constant in upper case?

File: base/src/main/java/io/fabric8/launcher/base/JsonUtils.java [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
+
     private JsonUtils() {
```
Comment (ia3andy) [inline base/src/main/java/io/fabric8/launcher/base/JsonUtils.java]: Since we use different Json Libraries, maybe naming it Jacksons or JsonMappers?

**Comment by ia3andy** [comment]
@gastaldi my first guess was to agree with your last comment about separation. However, I find your current solution pretty clean and simple. It seems to me that `Projectile` is not a "data" object but more an object which keeps the runtime state and configuration during the process (it already contains a `Path` for example). 
The only drawback I see about keeping it in the `Projectile` is if it ever get serialized.

**Comment by edewit** [comment]
I must agree that SSE is all we need as we don't need bi directional communication. But websockets are better supported in browsers and it's easier to detect if the browser is still connected. What would be the benefits of changing to SSE?

File: base/src/main/java/io/fabric8/launcher/base/JsonUtils.java (hunk 2 above)
Comment (gastaldi) [reply base/src/main/java/io/fabric8/launcher/base/JsonUtils.java]: That's an implementation detail, I think we should leave in the JsonUtils class for now and then separate later if needed

**Pull request merged**